kafka-init/src/main/java/io/strimzi/kafka/init/InitWriter.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    /**
+     * Write the rack-id
+     *
+     * @return if the operation was executed successfully
+     */
+    public boolean writeRack() {
+
+        Map<String, String> nodeLabels = client.nodes().withName(config.getNodeName()).get().getMetadata().getLabels();
+        log.info("NodeLabels = {}", nodeLabels);
+        String rackId = nodeLabels.get(config.getRackTopologyKey());
+        log.info("Rack: {} = {}", config.getRackTopologyKey(), rackId);
+
+        if (rackId == null) {
+            log.error("Node {} doesn't have the label {} for getting the rackid",
+                    config.getNodeName(), config.getRackTopologyKey());
+            return false;
+        }
+
+        return write(FILE_RACK_ID, rackId);
+    }
+
+    /**
+     * Write the external address of this node
+     *
+     * @return if the operation was executed successfully
+     */
+    public boolean writeExternalAddress() {
+
+        List<NodeAddress> addresses = client.nodes().withName(config.getNodeName()).get().getStatus().getAddresses();
+        log.info("NodeLabels = {}", addresses);
+        String externalAddress = findAddress(addresses);
+
+        if (externalAddress == null) {
+            log.error("External address nto found");
+            return false;
+        } else  {
+            log.info("External address found {}", externalAddress);
+        }
+
+        return write(FILE_EXTERNAL_ADDRESS, externalAddress);
+    }
+
+    /**
+     * Write provided information into a file
+     *
+     * @param file          Target file
+     * @param information   Information to be written
+     * @return              true if write succeeded, false otherwise
+     */
+    private boolean write(String file, String information) {
+        PrintWriter writer = null;
+        boolean isWritten;
+        try {
+            writer = new PrintWriter(config.getInitFolder() + "/" + file, "UTF-8");
+            writer.write(information);
+            log.info("Information {} written successfully to file {}", information, file);
+            isWritten = true;
+        } catch (IOException e) {
+            log.error("Error writing the information {} to file {}", information, file, e);
+            isWritten = false;
+        } finally {
+            if (writer != null)
+                writer.close();
+        }
+
+        return isWritten;
+    }
+
+    /**
+     * Tries to find the right address of the node. The different addresses has different prioprities:
+     *      1. ExternalDNS
+     *      2. ExternalIP
+     *      3. Hostname
+     *      4. InternalDNS
+     *      5. InternalIP
+     *
+     * @param addresses List of addresses which are assigned to our node
+     * @return  Address of the node
+     */
+    protected String findAddress(List<NodeAddress> addresses)   {
+        if (addresses == null)  {
+            return null;
+        }
+
+        NodeAddress found = null;
+
+        found = findAddressType(addresses, "ExternalDNS");
```

Comment:
These multiple calls to `findAddressType` are inefficient (OK, so it's not exactly perf sensitive code), but I also wonder about the possibility of multiple addresses with different types.

Reply:
Actually, this is expected that the node will contain multiple addresses., So the order of the `findAddressType` calls basically defines the priority with which we handle these. 

Performance IMHO doesn't really matter in the init container. And I expect the list of addresses to have usually just single digit number of items. But I'm fine with optimising this. But what would be a better way? Should we convert the list into a map with the address types as the key and afterwards do lookup for the key? That should be just one loop of the list and lookups in HashMap. Any better way?

Reply:
Ok, I will fix it.